### DISCUSSION TITLE
How about scripting?

### DISCUSSION THREAD

**Discussion by traverseda** [spacedriveapp/spacedrive#65] [Feature Request]
A good personal distributed filesystems is 99% of the way towards being a personal distributed *operating system*. Being able to upload scripts and schedule them to be run on certain files (inotify-like) or schedules (cron-like) or possibly in response to user events (gui-like) would let this become a personal operating system pretty easily. Of course exactly what that run-time looks like is a hard question, but it's something to consider.

Maybe extensions can be just files that you load dynamically?

**Comment by oscartbeaumont** [comment]
Extensions will be able to create their own custom jobs. These jobs will be scheduled between cores to execute. Hence this functionality is already planned and being built into the app. I think cron support is currently required for us internally but I think events on file update and GUI events would be a nice feature in the future.
